### PR TITLE
use mpicc to compile the code when gasnet is enabled

### PR DESCRIPTION
**Pull request by eddy16112**

### CODE REVIEW

File: src/legate.mk [hunk 1]
```diff
@@ -35,10 +35,37 @@ endif
 
 RM	:= rm
 
+# use mpi{cc,cxx,f90} compiler wrappers if USE_MPI=1 and we're not on a Cray system
```
Comment (magnatelee) [inline src/legate.mk:38]: `USE_MPI` should be `USE_GASNET`

Comment (eddy16112) [reply src/legate.mk]: Thanks, fixed now.

**Review by magnatelee — APPROVED** [review]

File: src/legate.mk [hunk 2]
```diff
@@ -35,10 +35,37 @@ endif
...
-INC_FLAGS += -I$(MPI_HOME)/include
-LD_FLAGS += -L$(MPI_HOME)/lib -lmpi
+  ifeq (${CRAYPE_VERSION},)
+    # OpenMPI check
+    ifneq ($(strip $(shell __INTEL_POST_CFLAGS+=' -we10006' $(CC) -showme:compile 2>&1 > /dev/null; echo $$?)),0)
+      # MPICH check
+      ifneq ($(strip $(shell __INTEL_POST_CFLAGS+=' -we10006' $(CC) -show 2>&1 > /dev/null; echo $$?)),0)
+        export OMPI_CC  	:= $(CC)
+        export MPICH_CC  	:= $(CC)
+        CC					:= mpicc
```
Comment (manopapad) [inline src/legate.mk:47]: Can you fix the alignment here?

Comment (eddy16112) [reply src/legate.mk]: good catch, somehow the alignment is OK in my vscode. Is there a way to let pre-commit catch the format issue of makefile?

Comment (manopapad) [reply src/legate.mk]: I think we'll be moving away from GNU Make soon, so I wouldn't worry about this, but maybe we should consider adding auto-formatting for our upcoming cmakefiles @trxcllnt @marcinz ?

**Pull request merged**